### PR TITLE
Add audit activities input plugin and mongo store support

### LINKED ISSUE #47: Support audit activity logs
**Is your feature request related to a problem? Please describe.**
Currently, all logs that are sent to Log Export Container are related to SDM Resources, but, there are some logs like Activities (that comes from SDM Audit) that would be great to see in LEC as well.

**Describe the solution you'd like**
Add an input plugin to execute, periodically, following the script from [StrongDM S3 Docs](https://www.strongdm.com/docs/installation/configure-logging/aws-s3), every 15 minutes, to send to LEC the SDM Audit Activity logs.

**Additional context**
It would be great to store the activity logs in a mongo instance like the last implementation about storing the common logs in Mongo (#45).

### PR DESCRIPTION
**Pull request by wallrony**
Closes #47

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,10 +12,22 @@ COPY Gemfile /Gemfile
...
+RUN rm sdm*.zip
+RUN mv sdm /home/fluent
+
 COPY fluentd /fluentd
 COPY create-conf-file.sh /create-conf-file.sh
 COPY start.sh /start.sh
 RUN chown fluent.fluent fluentd/etc/fluent.conf
 
-USER fluent
+RUN chown fluent.fluent /home/fluent/sdm
```
Comment (camposer) [inline Dockerfile:30]: If you're using _root_ you don't need this line.

File: Dockerfile [hunk 2]
```diff
@@ -12,10 +12,22 @@ COPY Gemfile /Gemfile
 COPY Gemfile.lock /Gemfile.lock
 RUN bundle install
 
+RUN apk --no-cache add curl unzip jq ca-certificates wget
+RUN wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
+RUN wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.28-r0/glibc-2.28-r0.apk
+RUN apk add glibc-2.28-r0.apk
```
Comment (camposer) [inline Dockerfile:18]: I think that you can replace this line by: `apk add libc6-compat`. See this dockerfile:

```
FROM alpine:latest

ENV SDM_DISABLE_UPDATE=true
ENV SDM_DOCKERIZED=true
ENV SDM_HOME=/home/sdm/.sdm/
ENV PATH=$PATH:/home/sdm

RUN apk --no-cache add curl libc6-compat

# This block can be replaced by a COPY/ADD command
# Download sdm for always using the latest client
RUN curl -J -O -L https://app.strongdm.com/releases/cli/linux
RUN unzip -x sdm*.zip
RUN rm sdm*.zip
RUN apk del curl

RUN adduser -D sdm
RUN mv sdm /home/sdm

COPY entrypoint.sh /entrypoint.sh
RUN chmod +x /entrypoint.sh

# Key! Avoid root access
USER sdm 

ENTRYPOINT ["/entrypoint.sh"]
```

File: create-conf-file.sh [hunk 3]
```diff
@@ -40,6 +40,15 @@ get_input_chunk_conf() {
   fi
 }
 
+get_input_exec_audit() {
+  exec_audit_enabled=$(echo $LOG_EXPORT_CONTAINER_AUDIT_ACTIVITIES | tr '[:upper:]' '[:lower:]')
```
Comment (camposer) [inline create-conf-file.sh:44]: Unused variable

File: Dockerfile (hunk 1 above)
Comment (camposer) [inline Dockerfile:30]: If you're using `root` you don't need this line.

File: create-conf-file.sh [hunk 4]
```diff
@@ -40,6 +40,15 @@ get_input_chunk_conf() {
...
+  if [ "$LOG_EXPORT_CONTAINER_AUDIT_ACTIVITIES" == "true" ]; then
```
Comment (camposer) [inline create-conf-file.sh:45]: We should evaluate here case insensitive values, e.g. TrUe

File: create-conf-file.sh [hunk 5]
```diff
@@ -71,6 +80,7 @@ get_output_conf() {
 
 create_fluent_conf() {
     cat $(get_input_conf_filename) > $ETC_DIR/fluent.conf
+    cat $(get_input_exec_audit) >> $ETC_DIR/fluent.conf
```
Comment (camposer) [inline create-conf-file.sh:83]: Name the variable and everything to be: extract audit activities

**Review by camposer — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (wallrony) [reply Dockerfile:18]: When using only `libc6-compat`, we get the following errors:

```
log-export-container_1  | Error relocating ./sdm: __vfprintf_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memcpy_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memset_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __fprintf_chk: symbol not found
log-export-container_1  | Creating Fluentd conf file
log-export-container_1  | Error relocating ./sdm: __vfprintf_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memcpy_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memset_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __fprintf_chk: symbol not found
```

We tried other approaches like using `gcompat` and other `glib` libraries, but it continues to display some errors meaning that it's still missing something.

Trying to run the dockerfile that you shared above in my Mac M1 Big Sur, it gives the following errors:

Without using DOCKER_DEFAULT_PLATFORM:

```
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

Using DOCKER_DEFAULT_PLATFORM=linux/amd64:

```
Error relocating /home/sdm/sdm: __vfprintf_chk: symbol not found
Error relocating /home/sdm/sdm: __memcpy_chk: symbol not found
Error relocating /home/sdm/sdm: __memset_chk: symbol not found
Error relocating /home/sdm/sdm: __fprintf_chk: symbol not found
```

Comment (camposer) [reply Dockerfile:18]: Try building in M1 with:
```
docker buildx build --platform linux/amd64 -t log-export-container .
```

Comment (wallrony) [reply Dockerfile:18]: It returns the same error:

```
Starting SDM
Error relocating ./sdm: __vfprintf_chk: symbol not found
Error relocating ./sdm: __memcpy_chk: symbol not found
Error relocating ./sdm: __memset_chk: symbol not found
Error relocating ./sdm: __fprintf_chk: symbol not found
```

Comment (wallrony) [reply Dockerfile:18]: It returns the same error:

```
Starting SDM
Error relocating ./sdm: __vfprintf_chk: symbol not found
Error relocating ./sdm: __memcpy_chk: symbol not found
Error relocating ./sdm: __memset_chk: symbol not found
Error relocating ./sdm: __fprintf_chk: symbol not found
```

Comment (camposer) [reply Dockerfile:18]: Perfect, merging then. Thank you guys

**Review by camposer — APPROVED** [review]

**Pull request merged**